server/game/core/ability/PlayCardAction.ts

```diff
@@ -62,4 +77,14 @@ export abstract class PlayCardAction extends PlayerAction {
         const resourceCost = this.cost.find((cost) => cost.getAdjustedCost);
         return resourceCost ? resourceCost.getAdjustedCost(context) : 0;
     }
+
+    public handleSmuggle(context: PlayCardContext) {
+        if (this.playType === PlayType.Smuggle) {
+            context.game.openEventWindow([
```

Comment:
Sorry, I don't know how I missed this before - since the new resource is considered to enter play simultaneously with the old one leaving per the rules, this event needs to go into the same window as the event that moves the smuggle card into play (or discard for events).

That should be pretty straightforward, just convert this to be like `generateSmuggleResourceEvent` or something and have it return the event, which you then insert into the window with the event that moves the card. Similar to how the putIntoPlay and cardPlayedEvent are emitted together for units, just insert the smuggle event into that same window with them.

For units and upgrades it's very obvious where the window that moves the card is being created since it's in the ctor, for event cards it happens here: https://github.com/SWU-Karabast/forceteki/blob/9c391fe88a42113edcc91a31af2bb15fbc0ffb4d/server/game/core/gameSteps/AbilityResolver.js#L259

Lmk if that needs additonal explanation, I think the overall code change should be relatively small.